module/VuFind/src/VuFind/RecordDataFormatter/Specs/DefaultRecord.php

```diff
@@ -0,0 +1,396 @@
+            [$this, 'getDefaultCollectionInfoSpecs']
+        );
+        $this->setDefaults(
+            'collection-record',
+            [$this, 'getDefaultCollectionRecordSpecs']
+        );
+        $this->setDefaults('core', [$this, 'getDefaultCoreSpecs']);
+        $this->setDefaults('description', [$this, 'getDefaultDescriptionSpecs']);
+    }
+
+    /**
+     * Get the callback function for processing authors.
+     *
+     * @return callable
+     */
+    public function getAuthorFunction(): callable
+    {
+        return function ($data, $options) {
+            // Lookup array of singular/plural labels (note that Other is always
+            // plural right now due to lack of translation strings).
+            $labels = [
+                'primary' => ['Main Author', 'Main Authors'],
+                'corporate' => ['Corporate Author', 'Corporate Authors'],
+                'secondary' => ['Other Authors', 'Other Authors'],
+            ];
+            // Lookup array of schema labels.
+            $schemaLabels = [
+                'primary' => 'author',
+                'corporate' => 'creator',
+                'secondary' => 'contributor',
+            ];
+
+            // Sort the data:
+            $final = [];
+            foreach ($data as $type => $values) {
+                $final[] = [
+                    'label' => $labels[$type][count($values) == 1 ? 0 : 1],
+                    'values' => [$type => $values],
+                    'options' => [
+                        'pos' => $options['pos'] + $this->authorOrder[$type],
+                        'renderType' => 'RecordDriverTemplate',
+                        'template' => 'data-authors.phtml',
+                        'context' => [
+                            'type' => $type,
+                            'schemaLabel' => $schemaLabels[$type],
+                            'requiredDataFields' => [
+                                ['name' => 'role', 'prefix' => 'CreatorRoles::'],
+                            ],
+                        ],
+                    ],
+                ];
+            }
+            return $final;
+        };
+    }
+
+    /**
+     * Get the settings for formatting language lines.
+     *
+     * @return array
+     */
+    public function getLanguageLineSettings(): array
+    {
+        if ($this->schemaOrgHelper) {
+            $langSpan = $this->schemaOrgHelper
+                ->getTag('span', ['property' => 'availableLanguage', 'typeof' => 'Language']);
+            $nameSpan = $this->schemaOrgHelper->getTag('span', ['property' => 'name']);
+            $itemPrefix = $langSpan . $nameSpan;
+            $itemSuffix = ($nameSpan ? '</span>' : '') . ($langSpan ? '</span>' : '');
+        } else {
+            $itemPrefix = $itemSuffix = '';
+        }
+        return compact('itemPrefix', 'itemSuffix') + [
+                'translate' => true,
+                'translationTextDomain' => 'ISO639-3::',
+            ];
```

Comment:
I don't think this needs to be indented quite so far:

```suggestion
        return compact('itemPrefix', 'itemSuffix') + [
            'translate' => true,
            'translationTextDomain' => 'ISO639-3::',
        ];
```